scipy/signal/signaltools.py

```diff
@@ -463,6 +460,9 @@ def convolve(in1, in2, mode='full'):
     volume = asarray(in1)
     kernel = asarray(in2)
 
+    if not _is_valid_mode_shapes(mode, volume.shape, kernel.shape):
+        volume, kernel = kernel, volume
```

Comment:
I suppose we could, but in the general case, having to check whether the dimensions "violate" regardless of order will slow things down IMHO.  Besides, the output is still computable even when they're no longer valid.  You just get empty arrays because every number in the output requires padded zeros.